e2e/test_sanity.py

```diff
@@ -0,0 +1,47 @@
+from playwright.sync_api import Page, expect
+
+import os
+import time
+
+BASE_URL = os.environ.get("BASE_URL", "http://localhost:8080/")
+
+
+def test_ingest_odt(page: Page) -> None:
+    """
+    Creates an investigation, uploads an odt file and awaits it to be there.
+    Searches for a word inside it, opens the file preview, looks at the extracted
+    text and browses to the last page.
+    """
+    investigation_name = "[test] Don't look in here"
+
+    page.goto(BASE_URL)
+    page.get_by_role("menuitem", name="Investigations").click()
+    page.get_by_role("button", name="New investigation").click()
+    page.get_by_placeholder("Untitled investigation").click()
+    page.get_by_placeholder("Untitled investigation").fill(investigation_name)
+    page.get_by_role("button", name="Save").click()
+    page.get_by_role("button", name="Upload documents").click()
+    page.get_by_label("Choose files to upload...").set_input_files(
+        "e2e/fixtures/random.odt"
+    )
+    page.get_by_role("button", name="Upload", exact=True).click()
+    page.get_by_role("dialog", name="Upload documents").get_by_role("button", name="Close").click()
```

Comment:
@tillprochaska I committed your more precise suggestion, but it seems to be less specific now, the test fails with: 

```
E       playwright._impl._api_types.Error: Error: strict mode violation: get_by_role("dialog", name="Upload documents").get_by_role("button", name="Close") resolved to 2 elements:
E           1) <button type="button" aria-label="Close" class="bp4-butt…>…</button> aka get_by_role("dialog", name="Upload documents").locator("div").filter(has_text="Upload documents").get_by_role("button", name="Close")
E           2) <button type="submit" class="bp4-button bp4-intent-prima…>…</button> aka locator("section").get_by_role("button", name="Close")
E
```

Would you ok with me rolling this change back?

Reply:
So the dialog actually has two close buttons:

![image](https://user-images.githubusercontent.com/1512805/231758713-640a7b0f-22f3-4a88-8f28-d4b931a3fc98.png)

So maybe try this?

```suggestion
    page.get_by_role("dialog", name="Upload documents").get_by_role("button", name="Close").first.click()
```

Reply:
Just to clarify why I’d prefer this over just a `section` selector, this is the section element in the dialog:

![image](https://user-images.githubusercontent.com/1512805/231760459-20842f94-5e00-4f94-b533-eb0ad8e20082.png)

To be honest, I’m not entirely sure if it’s semantically appropriate to use a section element in this case (that’s another question), but it’s something that could likely change when we work on the dialog itself to improve the UI or when we need to upgrade Blueprint (the UI component library we’re using).

Apart from that, as `section` is a very generic HTML element, there’s a risk of false positives and this would also match any other close button within a section element, so if for example we added a new section to the investigation homepage and then displayed a popover with a close button to promote the new feature, it would falsely match the locator.

Reply:
Understood and agreed. I generally use the codegen tool and this is what it spat out. But overall this is good, because it's showing us where we could be more specific in our markup, right?

Reply:
> But overall this is good, because it's showing us where we could be more specific in our markup, right?

Yes, absolutely, that’s also why I added it to the agenda for our discussion next week. Basically, I think there are three ways to handle situations like this:

* Use whatever locator works at the moment or the codgen tool generates, even if it’s not perfect, but maybe add a comment to the test or track this in an issue. In the manual feature descriptions/tests I started some time ago, you’ll often see things like *"Click on link with checkmark icon (should have invisible label, should be button or radio input)"*.

* Fix the issue at the same time as you add a new E2E test. This would be cool, but there may be situations where that’s not possible because of priorities/time or because it would require more complex refactorings. In this case, a fix would be to add an label to the button that’s visually hidden, but still accessible to Playwright and assistive tech like screen readers etc.

* Similar to the previous option, but don’t bother with fixing the underlying issue properly, and simply add a `data-test-id` or `data-qa` attribute to the element and use that in tests.